crates/compression/src/tables.rs

```diff
@@ -141,47 +230,59 @@ macro_rules! tables {
                         return Ok(RegistryKey::ZERO);
                     }
                     if let Some(found) = ctx.db.registry_index_lookup(self)? {
-                        ctx.accessed_keys[RegistryKeyspace::[<$type>]].insert(found);
+                        ctx.accessed_keys.$ident.insert(found);
                     }
                     Ok(RegistryKey::ZERO)
                 }
             }
 
-            impl<D: CompressDb> CompressibleBy<CompressCtx<D>> for $type {
+            impl<D> CompressibleBy<CompressCtx<D>> for $type
+            where
+                D: CompressDb
+            {
                 async fn compress_with(
                     &self,
                     ctx: &mut CompressCtx<D>,
                 ) -> anyhow::Result<RegistryKey> {
-                    if *self == Default::default() {
+                    if self == &Default::default() {
                         return Ok(RegistryKey::DEFAULT_VALUE);
                     }
                     if let Some(found) = ctx.db.registry_index_lookup(self)? {
                         return Ok(found);
                     }
 
-                    let key = ctx.per_keyspace.[<$type>].cache_evictor.next_key(&mut ctx.db)?;
-                    let old = ctx.per_keyspace.[<$type>].changes.insert(key, self.clone());
+                    let key = ctx.per_keyspace.$ident.cache_evictor.next_key(&mut ctx.db)?;
+                    let old = ctx.per_keyspace.$ident.changes.insert(key, self.clone());
                     assert!(old.is_none(), "Key collision in registry substitution");
                     Ok(key)
                 }
             }
 
-            impl<D: DecompressDb> DecompressibleBy<DecompressCtx<D>> for $type {
+            impl<D> DecompressibleBy<DecompressCtx<D>> for $type
+            where
+                D: DecompressDb
+            {
                 async fn decompress_with(
                     key: RegistryKey,
                     ctx: &DecompressCtx<D>,
                 ) -> anyhow::Result<Self> {
                     if key == RegistryKey::DEFAULT_VALUE {
                         return Ok(<$type>::default());
                     }
-                    ctx.db.read_registry(key)
+                    ctx.db.read_registry(&key)
                 }
             }
         )*
     }};
 }
 
-tables!(Address, AssetId, ContractId, ScriptCode, PredicateCode,);
+tables!(
+    address: Address,
+    asset_id: AssetId,
+    contract_id: ContractId,
+    script_code: ScriptCode,
+    predicate_code: PredicateCode
+);
```

Comment:
I removed this so the macros are easier to work with. Do you think it makes sense to introduce the `snake_case` names just to get rid of the `#[allow(non_snake_case)]`? Or is there another reason?